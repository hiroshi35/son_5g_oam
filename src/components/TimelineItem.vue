<template>
  <li :class="{'timeline-inverted': inverted}">
    <div :class="'timeline-badge ' + badgeColor">
      <div class="icon" v-html="badgeIcon"></div>
      <div class="date">
        <small v-text="date"></small>
      </div>
    </div>
    <div class="timeline-panel">
      <div class="timeline-heading">
        <h3 class="timeline-title" v-text="title"></h3>
      </div>
      <div class="timeline-body">
        <slot></slot>
      </div>
    </div>
  </li>
</template>

<script>
  export default {
    props: {
      title: {
        type: String
      },
      date: {
        type: String
      },
      badgeColor: {
        type: String,
        default: 'blue'
      },
      badgeIcon: {
        type: String,
        default: '<i class="fa fa-clock-o" aria-hidden="true"></i>'
      },
      inverted: {
        type: Boolean,
        default: false
      }
    }
  }
</script>
