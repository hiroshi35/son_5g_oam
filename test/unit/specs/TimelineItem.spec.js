import { shallow } from 'vue-test-utils'
import TimelineItem from '@/components/TimelineItem.vue'

describe('TimelineItem.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(TimelineItem,
      {
        propsData: {
          title: 'Item',
          date: '2 hours ago',
          badgeColor: 'blue',
          badgeIcon: '<i class="fa fa-clock-o" aria-hidden="true"></i>',
          inverted: false
        },
        slots: {
          default: 'Test'
        }
      })
  })
  it('should be vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('has correct props', () => {
    expect(wrapper.props().title).toBe('Item')
    expect(wrapper.props().date).toBe('2 hours ago')
    expect(wrapper.props().badgeColor).toBe('blue')
    expect(wrapper.props().badgeIcon).toBe('<i class="fa fa-clock-o" aria-hidden="true"></i>')
    expect(wrapper.props().inverted).toBe(false)
  })
  test('badge has correct class', () => {
    expect(wrapper.find('.timeline-badge ').classes()).toContain('blue')
  })
  test('badge has correct icon', () => {
    expect(wrapper.find('.icon').element.innerHTML).toBe('<i class="fa fa-clock-o" aria-hidden="true"></i>')
  })
  test('item don not have inverted class', () => {
    expect(wrapper.classes()).not.toContain('timeline-inverted')
  })
  it('should have inverted class when inverted is true', () => {
    wrapper.setProps({inverted: true})
    expect(wrapper.classes()).toContain('timeline-inverted')
  })
  it('should have correct text', () => {
    expect(wrapper.find('.timeline-body').text()).toBe('Test')
  })
})
