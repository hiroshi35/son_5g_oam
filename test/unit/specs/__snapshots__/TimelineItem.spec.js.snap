// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineItem.vue has the expected html structure 1`] = `
<li
  class=""
>
  <div
    class="timeline-badge blue"
  >
    <div
      class="icon"
    >
      <i
        aria-hidden="true"
        class="fa fa-clock-o"
      />
    </div>
     
    <div
      class="date"
    >
      <small>
        2 hours ago
      </small>
    </div>
  </div>
   
  <div
    class="timeline-panel"
  >
    <div
      class="timeline-heading"
    >
      <h3
        class="timeline-title"
      >
        Item
      </h3>
    </div>
     
    <div
      class="timeline-body"
    >
      Test
    </div>
  </div>
</li>
`;
